crates/uv/src/commands/mod.rs

```diff
@@ -231,6 +231,12 @@ impl<'a> OutputWriter<'a> {
     /// Commit the buffer to the output file.
     async fn commit(self) -> std::io::Result<()> {
         if let Some(output_file) = self.output_file {
+            if let Some(parent_dir) = output_file.parent() {
+                if !parent_dir.as_os_str().is_empty() && !parent_dir.is_dir() {
+                    fs_err::tokio::create_dir(parent_dir).await?;
```

Comment:
Don't we need [`create_dir_all`](https://doc.rust-lang.org/beta/std/fs/fn.create_dir_all.html)?

Reply:
I think we should just unequivocally call `create_dir_all`.

Reply:
Changed it to `create_dir_all`

Reply:
It might make sense to skip `try_exists` as suggested by Charlie, it already has a behavior for existing directories and probably handles race conditions?

Also, I think you can skip the empty check

> If the empty path is passed to this function, it always succeeds without creating any directories.

Reply:
ah ok done